remotes/docker/resolver.go

```diff
@@ -543,10 +547,12 @@ func (r *request) do(ctx context.Context) (*http.Response, error) {
 	if err != nil {
 		return nil, err
 	}
-	req.Header = http.Header{} // headers need to be copied to avoid concurrent map access
-	for k, v := range r.header {
-		req.Header[k] = v
+
+	req.Header = r.header.Clone() // headers need to be copied to avoid concurrent map access
+	if req.Header == nil {
+		req.Header = http.Header{}
 	}
```

Comment:
More a preference thing, but I feel the reverse feels better. e.g. default value and copy if it's not nil.

```suggestion
	req.Header = http.Header{}
	if r.header != nil {
		req.Header = r.header.Clone() // headers need to be copied to avoid concurrent map access
	}
```

Reply:
@dcantah the current change is more consistent with some existing code and reduce one extra `http.Header{}` in the case of `r.header != nil`.

https://github.com/containerd/containerd/blob/099d2e7c764536dff39243af2a5e281e8dec38c9/remotes/docker/resolver.go#L475-L478

Maybe we can change both to something like:

```golang
var header http.Header // declare a var if necessary
if r.header != nil {
    header = r.header.Clone()
} else {
    header = http.Header{}
}
```

WDYT?

Reply:
Yea that works fine

Reply:
updated! PTAL @dcantah thanks 😎